adapters/clients/client.go

```diff
@@ -92,6 +94,7 @@ type retryer struct {
 	minBackOff  time.Duration
 	maxBackOff  time.Duration
 	timeoutUnit time.Duration
+	// maxNumRetries int // TODO try this
```

Comment:
**nitpick (if-minor):** could be removed if not needed

Reply:
Thanks! Had this sitting locally but not pushed, done